Blish HUD/Controls/TextInputBase.cs

```diff
@@ -0,0 +1,718 @@
+﻿/*
+ *  This code is heavily adapted from the Myra TextBox (https://github.com/rds1983/Myra/blob/a9dbf7a1ceedc19f9e416c754eaf38e89a89a746/src/Myra/Graphics2D/UI/TextBox.cs)
+ *
+ *  MIT License
+ *
+ *  Copyright (c) 2017-2020 The Myra Team
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Blish_HUD.Input;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using MonoGame.Extended.BitmapFonts;
+using Microsoft.Xna.Framework.Input;
+
+namespace Blish_HUD.Controls {
+
+    /// <summary>
+    /// The base of controls such as the <see cref="TextBox"/> or <see cref="MultilineTextBox"/> which accept text input from the user.
+    /// </summary>
+    public abstract class TextInputBase : Control {
+
+        private static readonly Logger Logger = Logger.GetLogger<TextInputBase>();
+
+        protected static readonly char[] WordSeperators = { ' ', '\n', '`', '~', '!', '@', '#', '%', '^', '&', '*', '(', ')', '-', '=', '+', '[', '{', ']', '}', '\\', '|', ';', ':', '\'', '"', ',', '.', '<', '>', '/', '?' };
+
+        #region Load Static
+
+        private static readonly Color _highlightColor;
+
+        static TextInputBase() {
+            _highlightColor = new Color(92, 80, 103, 150);
+        }
+
+        #endregion
+
+        protected const char NEWLINE = '\n';
```

Comment:
What about `Environment.Newline`?

Reply:
`Environment.NewLine` returns a `string` instead of `char` and on Windows is `\r\n` instead of just `\n` which is what the multiline control supports in this implementation.

Reply:
Can be resolved